crates/cast/src/starknet_commands/script/init.rs

```diff
@@ -0,0 +1,141 @@
+use anyhow::{anyhow, Context, Ok, Result};
+use camino::Utf8PathBuf;
+use std::fs;
+
+use cast::helpers::constants::SCRIPTS_DIR;
+use cast::helpers::response_structs::ScriptInitResponse;
+use cast::helpers::scarb_utils::get_cairo_version;
+use clap::Args;
+use indoc::{formatdoc, indoc};
+use scarb_metadata::ScarbCommand;
+
+#[derive(Args, Debug)]
+pub struct Init {
+    /// Name of a script to create
+    pub script_name: String,
+}
+
+pub fn init(init_args: &Init) -> Result<ScriptInitResponse> {
+    let script_root_dir_path = get_script_root_dir_path(&init_args.script_name)?;
+
+    init_scarb_project(&init_args.script_name, &script_root_dir_path)?;
+    add_dependencies(&script_root_dir_path)?;
+    modify_files_in_src_dir(&init_args.script_name, &script_root_dir_path)?;
+
+    Ok(ScriptInitResponse {
+        status: format!("Successfully initialized `{}`", init_args.script_name),
+    })
+}
+
+fn get_script_root_dir_path(script_name: &str) -> Result<Utf8PathBuf> {
+    let current_dir = std::env::current_dir()?;
+
+    let script_root_dir_path = current_dir
+        .file_name()
+        .and_then(|dir_name| dir_name.to_str())
+        .filter(|&dir_name| dir_name == SCRIPTS_DIR)
+        .map_or_else(
+            || current_dir.join(SCRIPTS_DIR).join(script_name),
+            |_| current_dir.join(script_name),
+        );
+
+    Utf8PathBuf::from_path_buf(script_root_dir_path)
+        .map_err(|_| anyhow!("Failed to obtain script root dir path"))
+}
+
+fn init_scarb_project(script_name: &str, script_root_dir: &Utf8PathBuf) -> Result<()> {
```

Comment:
does this need to be a separate function?

Reply:
I'm still not convinced, as this function does nothing more than wrapping ScarbCommand::new. This is not reused anywhere nor does it modify the arguments in any way. I get it's a matter of preference, I just have my doubts about splitting every single thing into its own function - sometimes it makes things too overwhelming and decreases the readability.

Of course all things we've mentioned here are probably more of a subjective matter. I am rather against such shredding into smaller functions, but am more than happy to discuss if you guys prefer it the other way.

Reply:
> sometimes it makes things too overwhelming and decreases the readability.

I would say it is exactly the opposite

Reply:
In order to move things forward let's resolve this discussion and leave your changes, as this is not something inherently wrong. I would like to take more time on defining one approach for the whole project, because I don't like that we have different approaches for different subcommands. This may, most probably, become a part of Cast Refactor milestone. Sounds good?